indexing-service/src/main/java/io/druid/indexing/overlord/setup/EqualDistributionWorkerSelectStrategy.java

```diff
@@ -47,7 +47,8 @@ public int compare(
               ImmutableWorkerInfo zkWorker, ImmutableWorkerInfo zkWorker2
           )
           {
-            int retVal = -Ints.compare(zkWorker2.getCurrCapacityUsed(), zkWorker.getCurrCapacityUsed());
+            int retVal = Ints.compare(zkWorker2.getWorker().getCapacity() - zkWorker2.getCurrCapacityUsed(),
```

Comment:
Suggested to factor `getWorker().getCapacity() - getCurrCapacityUsed()` as a method in `ImmutableWorkerInfo`, i. e. called "getAvailableCapacity". Then you can use
```
Comparator.comparing(ImmutableWorkerInfo::getAvailableCapacity).reversed()
    .thenComparing(zkWorker -> zkWorker.getWorker().getVersion())
```

Instead of anonymous Comparator subclass